public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -244,6 +244,18 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<AzureM
     });
   }
 
+  newGetMetricNamespaces(resourceUri: string) {
```

Comment:
Sarah and I had a discussion about this and think we can leave the `new{existing method name}` functions in this PR as is and come back to it in https://github.com/grafana/grafana/issues/46065 as the existing methods should eventually be deleted anyway since they deal with the legacy query format and eventually the files under the `NewMetricsQueryEditor` directory can be moved into the `MetricsQueryEditor` directory.

@sarahzinger feel free to clarify/correct anything about my comment if I left something out.